### PR TITLE
Add User Report

### PR DESCRIPTION
**Pull request by shivanshuag**
Issue #29
### CountAll
- [x] Info - Returns the number of users registered  
- [x] Fail - If not able to fetch users for some reason
### Count Blocked
- [x] Info - Number of blocked users
### RolesList
- [x] Info - List of roles other than `anonymous` and `authenticated` with the user count of each
### WhoIsNumberOne
- [x] Info - Name and email of user with uid = 1
- [x] Fail - No user with uid = 1
### BlockedNumberOne
- [x] Pass - User with uid=1 is not blocked
- [x] Fail - User with uid = 1 is blocked. `drush user-block 1`

### CODE REVIEW

**Comment by fluxsauce** [comment]
Merge conflict, please rebase.

File: Check/Users/RolesList.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+   * Implements \SiteAudit\Check\Abstract\getAction().
+   */
+  public function getAction() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\calculateScore().
+   */
+  public function calculateScore() {
+    $roles = array_keys(\Drupal::entityManager()->getListBuilder('user_role')->load());
+    foreach ($roles as $role) {
```
Comment (fluxsauce) [inline Check/Users/RolesList.php]: This is incredibly inefficient. Please use a `GROUP BY`. https://github.com/fluxsauce/site_audit/blob/7.x-1.x/Check/Users/RolesList.php#L56

**Comment by fluxsauce** [comment]
> CountAll  Fail - If not able to fetch users for some reason

Does not catch exception.

**Comment by fluxsauce** [comment]
> WhoIsNumberOne Fail - No user with uid = 1

Doesn't work; I manually edited the UIDs to be something other than 1. If there's a better way to test this, please detail it.

**Comment by fluxsauce** [comment]
Please resolve the merge conflict and highlighted issues, then remove the needs work label.

File: Check/Users/RolesList.php (hunk 1 above)
Comment (shivanshuag) [inline Check/Users/RolesList.php]: In drupal 8, there is no separate table for roles. Roles are stored as configuration in the `config` table along with all the other config entities of the drupal 8 site in the form of a string - `user.role.[role-name]`. The role in the `user__roles` table is stored just as `[role-name]`. I could not find a way to join the two tables. That is why I had to resort to this method.

Comment (shivanshuag) [inline Check/Users/RolesList.php]: Oh wait, I misunderstood what you are saying. Will do it by group by. :)

**Comment by shivanshuag** [comment]
> Doesn't work; I manually edited the UIDs to be something other than 1. If there's a better way to test this, please detail it.

I think the problem is with cache. i changed the uid of user with uid=1 and then cleared the caches. The check returned `UID #1 does not exist! This is a serious problem.`

**Comment by shivanshuag** [comment]
> ```
>  CountAll Fail - If not able to fetch users for some reason
> ```
> 
> Does not catch exception.

There are many places in the code where a sql query is done and no exception is cached. I don't see a case where exception can occur if one does not modify the schema of default drupal tables.

**Comment by fluxsauce** [comment]
> There are many places in the code where a sql query is done and no exception is cached.

This particular one is important in case the users table is missing, which I've seen on bad imports.

**Comment by fluxsauce** [comment]
Looks good!

**Pull request merged**